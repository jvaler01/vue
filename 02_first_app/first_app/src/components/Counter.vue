<template>
  <h2>Counter</h2>
  <hr>
  <p>10</p>
</template>

<script>
export default {
    name: "Counter"
}
</script>

<style>

</style>