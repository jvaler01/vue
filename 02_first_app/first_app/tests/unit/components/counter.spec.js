import { shallowMount } from '@vue/test-utils'
import Counter from '@/components/Counter.vue'

describe('Counter Component', () => {
  /*test('Match with the snapshot', () => {
    const wrapper = shallowMount( Counter );

    //Check the HTML with a snapshot. -u flag update the shanpshot
    expect( wrapper.html() ).toMatchSnapshot();
  })*/

  test('h2 default value = Counter', () => {
    const wrapper = shallowMount( Counter );

    expect(wrapper.find('h2').exists()).toBeTruthy()

    const h2Text = wrapper.find('h2').text();

    expect( h2Text ).toBe('Counter');
  })
})