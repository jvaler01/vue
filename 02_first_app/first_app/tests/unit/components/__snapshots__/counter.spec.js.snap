// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Match with the snapshot 1`] = `
<h2>Counter</h2>
<hr>
<p>30 <sup>2</sup> = 900</p>
<div><button>+</button><button>-</button></div>
`;
